File: 001-BitwiseOperators/012.powerSetUsingBitwiseOperators.py
def powerSet(s):
    n = len(s)
    size = 2 ** n
    for i in range(0,size):
        for j in range(0,n):
            if i & (1 << j) != 0:
                print(s[j],end='')
        print()


powerSet('abc')

File: Geeks-For-Geeks/001-BitwiseOperators/012.powerSetUsingBitwiseOperators.js
const powerSet = (str) => {
  const n = str.length;
  const size = 2 ** n;
  const allSet = [];
  for (let i = 0; i < size; i++) {
    let subSet = '';
    for (let j = 0; j < n; j++) {
      if ((i & (1 << j)) != 0) {
        subSet += str[j];
      }
    }
    allSet.push(subSet);
  }
  console.log(allSet);
  return allSet;
};

powerSet('abc');
